common/lib/xmodule/xmodule/js/src/sequence/display.js

```diff
@@ -0,0 +1,354 @@
+                    window.clearTimeout(window.queuePollerID);
+                    delete window.queuePollerID;
+                }
+                this.render(newPosition);
+            } else {
+                alertTemplate = gettext('Sequence error! Cannot navigate to %(tab_name)s in the current SequenceModule. Please contact the course staff.');  // eslint-disable-line max-len
+                alertText = interpolate(alertTemplate, {
+                    tab_name: newPosition
+                }, true);
+                alert(alertText);  // eslint-disable-line no-alert
+            }
+        };
+
+        Sequence.prototype.selectNext = function(event) {
+            this._change_sequential('next', event);
+        };
+
+        Sequence.prototype.selectPrevious = function(event) {
+            this._change_sequential('previous', event);
+        };
+
+        // `direction` can be 'previous' or 'next'
+        Sequence.prototype._change_sequential = function(direction, event) {
+            var analyticsEventName, isBottomNav, newPosition, offset, widgetPlacement;
+
+            // silently abort if direction is invalid.
+            if (direction !== 'previous' && direction !== 'next') {
+                return;
+            }
+            event.preventDefault();
+            analyticsEventName = 'edx.ui.lms.sequence.' + direction + '_selected';
+            isBottomNav = $(event.target).closest('nav[class="sequence-bottom"]').length > 0;
+
+            if (isBottomNav) {
+                widgetPlacement = 'bottom';
+            } else {
+                widgetPlacement = 'top';
+            }
+
+            // Formerly known as seq_next and seq_prev
+            Logger.log(analyticsEventName, {
+                id: this.id,
+                current_tab: this.position,
+                tab_count: this.num_contents,
+                widget_placement: widgetPlacement
+            });
+
+            if ((direction === 'next') && (this.position >= this.contents.length)) {
+                window.location.href = this.nextUrl;
+            } else if ((direction === 'previous') && (this.position === 1)) {
+                window.location.href = this.prevUrl;
+            } else {
+                // If the bottom nav is used, scroll to the top of the page on change.
+                if (isBottomNav) {
+                    $.scrollTo(0, 150);
+                }
+
+                offset = {
+                    next: 1,
+                    previous: -1
+                };
+
+                newPosition = this.position + offset[direction];
+                this.render(newPosition);
+            }
+        };
+
+        Sequence.prototype.link_for = function(position) {
+            return this.$('#sequence-list .nav-item[data-element=' + position + ']');
+        };
+
+        Sequence.prototype.mark_visited = function(position) {
+            // Don't overwrite class attribute to avoid changing Progress class
```

Comment:
I don't know if this is a reference to that progress stuff I removed, or something else (it says "progress class", which might either refer to an html classname or another Coffeescript class somewhere?) I'd rather leave it since I don't know and its just a comment.